### PR TITLE
Feat(cx-1635): initial android push notification setup

### PR DESCRIPTION
**Pull request by lordkiz**
The type of this PR is: **Feature**

This PR resolves [CX-1635]

### Description

https://user-images.githubusercontent.com/18648835/124286024-cc82fa80-db4e-11eb-9a06-b62ad464ff94.mp4

#### Extra Details
- This PR specifically addresses android only.
- I have used the closest latest compatible firebase dep versions
- Required google play services. Using v 17.0.0. The implication of this is that other libs that uses the main definition of google play services version will now use v17.0.0. Therefore tipsi-stripe which was using v11.0.0 will now use v17.0.0. 
- firebase analytics was added too, so the notification reports such as clicks etc is visible on the firebase dashboard.
- We are using firebaseMessagingVersion 21.1.0. React-native-device-info uses firebaseIid. In order to prevent Duplicate class errors, I have upped firebaseIidVersion to 21.1.0

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Initial Push Notification Setup on android

[CX-1635]: https://artsyproduct.atlassian.net/browse/CX-1635

### CODE REVIEW

**Comment by lordkiz** [comment]
CI build for android will fail until build injection of google-services.json is set up. This is related to google auth that @edez1t is working on.
Screenshot 2021-07-02 at 17 12 42

File: src/lib/utils/AdminMenu.tsx [hunk 1]
```diff
@@ -180,6 +188,51 @@ const Buttons: React.FC<{ onClose(): void }> = ({ onClose }) => {
...
+        ])
+      }}
+      value={
+        <Text variant="subtitle" color="black60">
+          {flagValue}
+        </Text>
+      }
+    />
+  )
+}
```
Comment (brainbicycle) [inline src/lib/utils/AdminMenu.tsx:234]: 🤩 Nice addition!

**Review by brainbicycle — APPROVED** [review]
Nice work 💪!

**Comment by edez1t** [comment]
> Nice work ! This looks good to me to merge once the CI tests are passing.

Merge master to make it happen

File: src/lib/utils/AdminMenu.tsx (hunk 1 above)
Comment (olerichter00) [reply src/lib/utils/AdminMenu.tsx:234]: 🎉

**Review by olerichter00 — APPROVED** [review]
Good job! 🎉

**Pull request merged**